src/com/facebook/buck/android/AabBuilderStep.java

```diff
@@ -98,128 +96,132 @@ public StepExecutionResult execute(ExecutionContext context) throws IOException
       output = context.getStdOut();
     }
 
-    try {
-      AppBuilderBase.PrivateKeyAndCertificate privateKeyAndCertificate =
-          appBuilderBase.createKeystoreProperties();
-      File fakeResApk = filesystem.createTempFile("fake", ".txt").toFile();
-      fakeResApk.createNewFile();
-      ApkBuilder builder =
-          new ApkBuilder(
-              filesystem.getPathForRelativePath(pathToOutputApkFile).toFile(),
-              fakeResApk,
-              null,
-              privateKeyAndCertificate.privateKey,
-              privateKeyAndCertificate.certificate,
-              output);
-      builder.setDebugMode(debugMode);
-      Set<String> addedFiles = new HashSet<>();
-      Set<Path> addedSourceFiles = new HashSet<>();
-      for (ModuleInfo moduleInfo : modulesInfo) {
-        addModule(builder, moduleInfo, addedFiles, addedSourceFiles);
+    ImmutableList.Builder<Path> modulesPathsBuilder = new Builder<>();
+    File fakeResApk = filesystem.createTempFile("fake", ".txt").toFile();
+    Set<String> addedFiles = new HashSet<>();
+    Set<Path> addedSourceFiles = new HashSet<>();
+
+    for (ModuleInfo moduleInfo : modulesInfo) {
+      Path moduleGenPath = getPathForModule(moduleInfo);
+      StepExecutionResult moduleBuildResult =
+          addModule(
+              context, moduleGenPath, fakeResApk, output, moduleInfo, addedFiles, addedSourceFiles);
```

Comment:
`output` can be a `null` here, `addModule` need to have `@Nullable` on the corresponding argument